src/layer/LayersFactory.ts

```diff
@@ -293,4 +314,73 @@ export class LayersFactory {
         new WmsLayer({ baseUrl, layerId, title, description, legendUrl }),
     );
   }
+
+  private static async makeLayersWmts(
+    baseUrl: string,
+    filterLayers: Function | null,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    overrideConstructorParams: Record<string, any> | null,
+    reqConfig: RequestConfiguration,
+  ): Promise<AbstractLayer[]> {
+    const parsedLayers = await fetchLayersFromWmtsGetCapabilitiesXml(baseUrl, reqConfig);
+    const layersInfos = parsedLayers.map(layerInfo => ({
+      layerId: layerInfo.Name[0],
+      title: layerInfo.Title[0],
+      description: layerInfo.Abstract ? layerInfo.Abstract[0] : null,
+      dataset: null,
+      legendUrl: layerInfo.Style[0].LegendURL,
+      resourceUrl: layerInfo.ResourceUrl,
+    }));
+
+    const filteredLayersInfos =
+      filterLayers === null ? layersInfos : layersInfos.filter(l => filterLayers(l.layerId, l.dataset));
+
+    return filteredLayersInfos.map(
+      ({ layerId, title, description, legendUrl, resourceUrl }) =>
+        new WmtsLayer({ baseUrl, layerId, title, description, legendUrl, resourceUrl }),
+    );
+  }
+
+  // Analtyc layers accept a proc parameter to specify a dynamically-rendered false color visualization, for example NDVI.
+  // Since proc is not a standard a WMTS parameter and there is a list of specified options at https://developers.planet.com/docs/basemaps/tile-services/indices/#remote-sensing-indices-legends
+  // we can treat these as extra layers and add them to makeLayers response.
+  private static async makePlanetBasemapLayers(
+    baseUrl: string,
+    filterLayers: Function | null,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    overrideConstructorParams: Record<string, any> | null,
+    reqConfig: RequestConfiguration,
+  ): Promise<AbstractLayer[]> {
+    let newLayers = [];
+    const parsedLayers = await fetchLayersFromWmtsGetCapabilitiesXml(baseUrl, reqConfig);
+    for (let layerInfo of parsedLayers) {
+      const layer = {
+        layerId: layerInfo.Name[0],
+        title: layerInfo.Title[0],
+        description: layerInfo.Abstract ? layerInfo.Abstract[0] : null,
+        dataset: null as string | null,
```

Comment:
I don't quite understand what this is supposed to do tbh? 

Any way, we usually make a dataset object for collection-specific layers. I think it wouldn't hurt making it here, although admittedly there probably isn't much benefit to it. Just to be consistent.

Reply:
Added a dataset object